aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -545,6 +545,32 @@ void dot<at::Half>(CUDABLAS_DOT_ARGTYPES(at::Half)) {
 #endif
 }
 
+template <>
+void vdot<c10::complex<float>>(CUDABLAS_DOT_ARGTYPES(c10::complex<float>)) {
+  #ifndef __HIP_PLATFORM_HCC__
+  TORCH_CUDABLAS_CHECK(cublasCdotc(handle, n, reinterpret_cast<const cuComplex*>(x),
```

Comment:
can we just add a cuda to hip mapping for `cuComplex` -> `rocblas_float_complex` and `cuDoubleComplex` -> `rocblas_double_complex`?

cc. @jeffdaily

Reply:
Let's try it.  I didn't see cuComplex used in any other context besides blas.

Reply:
@kshitij12345 can you try add a mapping for `cuComplex` -> `rocblas_float_complex` in `torch/utils/hipify/cuda_to_hip_mappings.py`?

Reply:
Have tried. 

Added the following (however not sure if it is correct)
```
("cuComplex", ("rocblas_float_complex", CONV_TYPE, API_RUNTIME)),
("cuDoubleComplex", ("rocblas_double_complex", CONV_TYPE, API_RUNTIME)),
```

Verify with CI as I don't have the machinery to do ROCm build locally.

Reply:
LGTM.  CI build succeeded, but test jobs got assigned to a CI host we only recently found to be broken.  That host was just taken offline.  Let's try a pytorchbot retest to verify.

Reply:
I think it should be:
```
("cuComplex", ("rocblas_float_complex", CONV_TYPE, API_BLAS)),
("cuDoubleComplex", ("rocblas_double_complex", CONV_TYPE, API_BLAS)),
```

Reply:
That's fair, yes, perhaps to avoid issues in the future if we actually do any major filtering of these meta data tags like `API_BLAS`.  The only tags we actually check for in the hipify_python.py script are `API_CAFFE2` and `API_PYTORCH` to differentiate between substitutions that are specific to one or other framework.  Otherwise, we apply the given substitution as-is to all sources.

Reply:
Sure. Will do that.

However `cuComplex` is part of `cuComplex.h`, so should there be a API_COMPLEX or something similar going forward? Just curious to know more as I don't know much about the ROCm effort.

Thanks!